js/ai/src/model/middleware.ts

```diff
@@ -235,6 +240,171 @@ export function augmentWithContext(
   };
 }
 
+/**
+ * Options for the `retry` middleware.
+ */
+export interface RetryOptions {
+  /**
+   * The maximum number of times to retry a failed request.
+   * @default 3
+   */
+  maxRetries?: number;
+  /**
+   * An array of `StatusName` values that should trigger a retry.
+   * @default ['UNAVAILABLE', 'DEADLINE_EXCEEDED', 'RESOURCE_EXHAUSTED', 'ABORTED', 'INTERNAL']
+   */
+  statuses?: StatusName[];
+  /**
+   * The initial delay between retries, in milliseconds.
+   * @default 0
+   */
+  initialDelayMs?: number;
+  /**
+   * The factor by which the delay increases after each retry (exponential backoff).
```

Comment:
nit: seems odd that "exponential" backoff defaults to linear... maybe it should be 2 as a default?